gax/src/main/java/com/google/api/gax/rpc/testing/FakeStatusCode.java

```diff
@@ -0,0 +1,79 @@
+/*
```

Comment:
Why do we have so many `Fake*` classes in `main` (not `test`) directory? Are ther really necessary? Can't many of them be replaced by just mockito mocks? Can't they be in `test/` folder instead of `main/*/testing`?

Reply:
I believe the original reason was so that the classes could be used across package boundaries, e.g. in gax-grpc - there is currently at least one thing being referenced. It's possible some of them could be replaced by mockito mocks, but I think it's actually nice to implement the classes this way as a validation at the core level of the implementability of the interfaces.

Reply:
What about moving them in main/*/testing? They will still be accessible in tests across package boundaries. This is really ugly to have so much "test garbage" in production code/artifact.